### PR TITLE
Clean up all __init__

### PR DESCRIPTION
**Pull request by ramipellumbi**
**IMPORTANT: THE BASE __INIT__ NEEDS TO BE DONE BY SOMEBODY WHO KNOWS WHAT IS SUPPOSED TO BE PUBLICLY EXPORTED**

- Precursor to adding OLLAMA support
- Does a ruff check and format of the prior pr to fix import ordering and formatting (really need to set up a pre-lint)
- Removes all local package __init__ contents except the ones that exported something not a package module class

### CODE REVIEW

**Comment by mdr223** [comment]
Just a heads up that running `pytest -v tests/pytest/` on this currently yields:
```
ImportError while loading conftest '/Users/matthewrusso/tmp/palimpzest/tests/pytest/conftest.py'.
tests/pytest/conftest.py:3: in <module>
    from palimpzest.constants import Model
src/palimpzest/__init__.py:3: in <module>
    from .cost_model import *
src/palimpzest/cost_model/__init__.py:1: in <module>
    from .cost_model import CostModel
src/palimpzest/cost_model/cost_model.py:19: in <module>
    from palimpzest.datamanager import DataDirectory
src/palimpzest/datamanager/__init__.py:1: in <module>
    from .datamanager import DataDirectory
src/palimpzest/datamanager/datamanager.py:10: in <module>
    from palimpzest.datasources.datasources import (
src/palimpzest/datasources/__init__.py:1: in <module>
    from .datasources import (
src/palimpzest/datasources/datasources.py:18: in <module>
    from palimpzest.tools.pdfparser import get_text_from_pdf
src/palimpzest/tools/pdfparser.py:15: in <module>
    from palimpzest.datamanager.datamanager import DataDirectory
E   ImportError: cannot import name 'DataDirectory' from partially initialized module 'palimpzest.datamanager.datamanager' (most likely due to a circular import) (/Users/matthewrusso/tmp/palimpzest/src/palimpzest/datamanager/datamanager.py)
```
I think you may want to empty the contents of the `__init__.py` files for the sub-directories in order to avoid these errors (now that we're importing everything directly)

**Review by mdr223 — APPROVED** [review]
Left a couple comments, but all very minor. I went ahead and addressed the ones re: the `code_synthesis_convert.py` calls to the `CustomGenerator` likely needing `context=None` set explicitly.

I think the only one that you may want to consider acting on is potentially adding the sanity check for calls to `<generator>.generate()` to `convert.py` (and possibly `token_reduction_convert.py` as well).

Pre-emptively approving since tests are passing.

File: src/palimpzest/generators/generators.py [hunk 1]
```diff
@@ -164,13 +166,15 @@ def _get_answer_log_probs(self, dspy_lm: dsp.LM, answer: str) -> List[float]:
         after=log_attempt_number,
         reraise=True,
     )
-    def generate(self, prompt: str) -> GenerationOutput:
+    def generate(self, context, prompt, **kwargs) -> GenerationOutput:
```
[the remark this answers is not in the document]
Comment (mdr223) [reply src/palimpzest/generators/generators.py:169]: (went ahead and added `context=None` to lines 382 and 479 of `code_synthesis_convert.py`)

[the remark this answers is not in the document]
Comment (ramipellumbi) [reply src/palimpzest/generators/generators.py:169]: This was only ever invoked when the input was None for context -- you are right I did not catch those (yay for reviews)

**Pull request merged**